error/err.go

```diff
@@ -0,0 +1,13 @@
+package error
+
+import "fmt"
+
+// ConnectionError is an error in the connection to the server
+type ConnectionError struct {
+	Msg string
```

Comment:
Have we decided on the types of errors to expose?
I don't think the struct members should be public